### PR TITLE
Sample .editorconfig

### PR DESCRIPTION
**Pull request by williamhatcher**
Feel free to add onto this as y'all see fit.
This is only a starter set if you will.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+root = true
+
+[*]
+indent_style = true
+end_of_line = lf
```
Comment (madmaxoft) [inline .editorconfig]: (and indeed I get them right currently, because my git is configured properly)

Comment (williamhatcher) [inline .editorconfig]: Can we force pre-commit hooks?

Comment (williamhatcher) [inline .editorconfig]: Okay. Have we discussed making the same thing for all plugins/lua code included in the build?

Comment (bearbin) [inline .editorconfig]: @madmaxoft Can you override `.editorconfig` to use your own preferred settings on your editor? If you can do that then we can set `end_of_line=lf`. The editorconfig should only be a recommendation, with the final control up to the user

Comment (williamhatcher) [inline .editorconfig]: The docs say to leave values blank in these cases

Comment (bearbin) [inline .editorconfig]: @williamhatcher can you provide a link to the docs?

**Comment by williamhatcher** [comment]
Let's move the discussion out of that commit.
Here's the link to their wiki
https://github.com/editorconfig/editorconfig/wiki
@bearbin

**Comment by madmaxoft** [comment]
Note to MSVC users: Using MSVC17 together with this file causes MSVC to force-add UTF8 BOM to all edited files. You need to update to "Update 15.3" to fix the issue.

**Comment by williamhatcher** [comment]
Sorry, I don't use MSVC. Maybe put that note in the contributing file?

**Comment by madmaxoft** [comment]
Nah, most people do update their stuff. And if they don't and search for "UTF8 BOM" in here, they'll find the comment.

**Pull request merged**